### ISSUE TITLE
Need an exploit.lua library to unify exploit script interface

### ISSUE THREAD

**Issue by dmiller-nmap** [nmap/nmap#186]
Possible function: x

Possibly include some things like:

* function to return the value of the most-specific provided script-arg between "exploit.cmd" and  "SCRIPT_NAME.cmd".
* function to produce and detect a unique string on various architectures (e.g. PHP, Java, cmd.exe, /bin/sh), useful for confirming code execution.
* lists of known files to check for file inclusion or directory traversal bugs
* common output format like vulns.lua? Or just use vulns.lua.

**Comment by cldrn** [comment]
Hi Daniel,

What do you mean in number 1? Implementing logic that will select a value if BOTH arguments were set?

**Comment by dmiller-nmap** [comment]
@cldrn Yes, a way to use `--script-args 'exploit.cmd="uname -a", vuln-script-x.cmd="whoami"'` and get the right thing.

**Comment by h4ck3rk3y** [comment]
i had a similar idea regarding directory traversal and lfi scripts earlier. a lot of them have similar code we could use an exploit library. is anyone assigned to this?

**Comment by p-l-** [comment]
@cldrn that's what is implemented for the `ls` module proposal (see #106).
